src/containers/CheckoutPage/CheckoutPage.duck.js

```diff
@@ -177,9 +184,34 @@ export const initiateOrder = (orderParams, transactionId) => (dispatch, getState
     expand: true,
   };
 
-  const createOrder = transactionId ? sdk.transactions.transition : sdk.transactions.initiate;
+  if (!transactionId) {
+    return initiatePrivileged({ isSpeculative: false, bookingData, bodyParams, queryParams })
+      .then(response => {
+        const entities = denormalisedResponseEntities(response);
+        const order = entities[0];
+        dispatch(initiateOrderSuccess(order));
+        dispatch(fetchCurrentUserHasOrdersSuccess(true));
+        return order;
+      })
+      .catch(e => {
+        dispatch(initiateOrderError(storableError(e)));
+        const transactionIdMaybe = transactionId ? { transactionId: transactionId.uuid } : {};
+        log.error(e, 'initiate-order-failed', {
+          ...transactionIdMaybe,
+          listingId: orderParams.listingId.uuid,
+          bookingStart: orderParams.bookingStart,
+          bookingEnd: orderParams.bookingEnd,
+        });
+        throw e;
+      });
+  }
 
-  return createOrder(bodyParams, queryParams)
+  return transitionPrivileged({
+    isSpeculative: false,
+    bookingData,
+    bodyParams,
+    queryParams,
+  })
```

Comment:
This might the style that is used in FTW widely, in which case you can just ignore my comment, but I find the early return in if, followed by implicit else that also returns, very confusing to read. So technically else is not needed here because if returns but for clarity I'd put this in an else block.

Reply:
Yeah, this is one of those things that yield debate. We used to have a lint rule that enforced one over the other (don't remember which), but at least at some point I think Prettier rewrote some of those else statements away.